packages/babel-helpers/src/helpers.js

```diff
@@ -461,6 +461,8 @@ helpers.set = template(`
 
       if (parent !== null) {
         set(parent, property, value, receiver);
+      } else {
+        receiver[property] = value;
       }
     } else if ("value" in desc && desc.writable) {
       desc.value = value;
```

Comment:
Yes, the silent fail in sloopy mode or throw in strict mode for `writeable: false` and no setter is a hard problem, that's why I do not want to deal with it in this PR. Let's just do one thing in one time.